tests/trainer/test_trainer.py

```diff
@@ -2951,6 +2956,16 @@ def test_accelerator_config_only_deprecated_args(self):
                 trainer = Trainer(model=model, args=args, eval_dataset=eval_dataset)
                 self.assertEqual(trainer.accelerator.split_batches, True)
 
+    def test_accelerator_custom_state(self):
+        AcceleratorState._reset_state(reset_partial_state=True)
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            with self.assertRaises(ValueError) as cm:
+                _ = RegressionTrainingArguments(output_dir=tmp_dir, accelerator_config={"use_configured_state": True})
+                self.assertIn("Please define this beforehand", str(cm.warnings[0].message))
+            _ = Accelerator()
+            _ = RegressionTrainingArguments(output_dir=tmp_dir, accelerator_config={"use_configured_state": True})
```

Comment:
Do we have to take the state from the global environment like this? It seems leaky and prone to possible issues. For example, how does this interplay with situations like multiprocessing? 

Could we instead pass in the state directly to the training arguments?

Reply:
It's not from a global environment, the state is a stateful object that can only ever be initialized once. It's perfectly safe in a threaded environment. 

Ideally yes, we pass in something like the `PartialState` or `Accelerator` directly in instead, but it is very safe

Reply:
OK - sounds good to me!